spacepy/__init__.py

```diff
@@ -95,49 +95,105 @@ def help():
     else:
         os.environ['PATH'] = minglibs
 
-def deprecated(version, message):
+#actual deprecation decorator
+def _deprecator(version, message, docstring, func):
+    #this is the actual, deprecated function
+    @functools.wraps(func)
+    def _deprecated(*args, **kwargs):
+        warnings.warn(message, DeprecationWarning)
+        return func(*args, **kwargs)
+    if func.__doc__ is None:
+        doclines = []
+    else:
+        doclines = func.__doc__.split('\n')
+    # Docstring SHOULD be a single non-blank line with summary, blank line,
+    # and then the rest of the content. Want to put the deprecation
+    # information just before the first line of "rest of content"
+    isblank = [not bool(l.strip()) for l in doclines]
+    # All places with a non-blank line following a blank
+    paragraphs =  [i for i in range(len(isblank))
+                   if not isblank[i] and
+                   (i == 0 or isblank[i - 1])]
+    if not paragraphs: # No non-blank, guess indentation, insert at end
+        leading = '    '
+        insert_at = len(doclines) #insert at end
+    elif len(paragraphs) == 1: # Get indent from only para, insert at end
+        l = doclines[paragraphs[0]]
+        leading = l[:len(l) - len(l.lstrip())]
+        insert_at = len(doclines)
+    else: # Get indent from 2nd paragraph, insert just before it
+        l = doclines[paragraphs[1]]
+        leading = l[:len(l) - len(l.lstrip())]
+        # Insert before blank line before the paragraph.
+        insert_at = paragraphs[1] - 1
+    to_insert = [
+        '',
+        leading + '.. deprecated:: ' + version,
+    ] \
+    + [leading + '   ' + d for d in docstring.split('\n')]
+    doclines[insert_at:insert_at] = to_insert
+    _deprecated.__doc__ = '\n'.join(doclines)
+    return _deprecated
+
+def deprecated(version, message, docstring=None):
     """Decorator to deprecate a function/method
 
+    Modifies a function so that calls to it raise
+    ``DeprecationWarning`` and the docstring has a deprecation
+    note added in accordance with `numpydoc format
+    <https://numpydoc.readthedocs.io/en/latest/format.html#sections>`_
+
     Parameters
     ==========
     version : str
         What is the first version where this was deprecated?
 
     message : str
-        Message to include in the documentation and the warning message.
+        Message to include in the deprecation warning and in the
+        docstring.
+
+    Other Parameters
+    ================
+    docstring : str
+        If specified, this will be used in the docstring instead of
+        ``message``. It can be a multi-line string (separated with
+        ``\\n``). It will be indented to match the existing docstring.
+
```

Comment:
It took a moment to realize the utility of `docstring` vs. `message`.  I recommend adding, "``message`` is still used in the warning message" or something similar in the docstring here.

Reply:
Tried to clarify this in the update; mark "resolve" if you're good with it.